crates/testing/tests/quorum_vote_task.rs

```diff
@@ -0,0 +1,143 @@
+#![allow(clippy::panic)]
+use hotshot::tasks::task_state::CreateTaskState;
+use hotshot_example_types::node_types::{MemoryImpl, TestTypes};
+use hotshot_types::{data::ViewNumber, traits::node_implementation::ConsensusTime};
+
+#[cfg(test)]
+#[cfg_attr(async_executor_impl = "tokio", tokio::test(flavor = "multi_thread"))]
+#[cfg_attr(async_executor_impl = "async-std", async_std::test)]
+async fn test_quorum_vote_task_success() {
+    use either::{Left, Right};
+    use hotshot::tasks::inject_quorum_vote_polls;
+    use hotshot_task_impls::{events::HotShotEvent::*, quorum_vote::QuorumVoteTaskState};
+    use hotshot_testing::{
+        predicates::{consecutive, exact},
+        script::{run_test_script, TestScriptStage},
+        task_helpers::build_system_handle,
+        view_generator::TestViewGenerator,
+    };
+
+    async_compatibility_layer::logging::setup_logging();
+    async_compatibility_layer::logging::setup_backtrace();
+
+    let handle = build_system_handle(2).await.0;
+    let quorum_membership = handle.hotshot.memberships.quorum_membership.clone();
+
+    let mut generator = TestViewGenerator::generate(quorum_membership.clone());
+
+    generator.next();
+    let view = generator.current_view.clone().unwrap();
+
+    // Send the quorum proposal, DAC, and VID disperse data, in which case a dummy vote can be
+    // formed and the view number will be updated.
+    let view_success = TestScriptStage {
+        inputs: vec![
+            QuorumProposalRecv(view.quorum_proposal.clone(), view.leader_public_key),
+            DACRecv(view.da_certificate.clone()),
+            VidDisperseRecv(view.vid_proposal.0.clone()),
+        ],
+        outputs: vec![
+            Right(consecutive((
+                ViewChange(ViewNumber::new(2)),
+                QuorumProposalValidated(view.quorum_proposal.data.clone()),
+            ))),
+            Right(consecutive((
+                DACertificateValidated(view.da_certificate.clone()),
+                VIDShareValidated(view.vid_proposal.0.data.clone()),
+            ))),
+            Left(exact(QuorumVoteDependenciesValidated(ViewNumber::new(1)))),
+            Left(exact(DummyQuorumVoteSend(ViewNumber::new(1)))),
```

Comment:
Nondeterministic issue: we don't always get both of these two events, and sometimes get neither. I'm wondering if this is because the quorum vote task was shut down before receiving these two tasks triggered by the completed dependency task.

Trying to disable the shutdown or add a pause doesn't help. I'll keep investigating, but please let me know if there's an obvious bug or solution.

Reply:
I managed to fix this by pausing. Let's discuss tomorrow.

Reply:
I replaced `try_recv` with `recv` and timeout as we discussed today.